snapshot/snapshot.go

```diff
@@ -325,7 +330,11 @@ func (o *snapshotter) Prepare(ctx context.Context, key, parent string, opts ...s
 		}
 		log.G(lCtx).WithField(remoteSnapshotLogKey, prepareFailed).WithError(err).Warn("failed to prepare remote snapshot")
 		if !errors.Is(err, ErrNoZtoc) {
-			commonmetrics.IncOperationCount(commonmetrics.FuseMountFailureCount, digest.Digest(""))
+			if errors.Is(err, ErrNoIndex) {
+				skipLazyLoadingImage = true
+			} else {
+				commonmetrics.IncOperationCount(commonmetrics.FuseMountFailureCount, digest.Digest(""))
+			}
```

Comment:
It might be good to refactor this error handling into a switch case with more clear logging.